### PR TITLE
Update modules to reflect DSL2

### PR DESCRIPTION
**Pull request by gregorysprenger**
Updated modules.md to reflect DSL2 with a warning about the end of life of DSL1. Also added basic information about what DSL means. This is related to issue https://github.com/nf-core/nf-co.re/issues/1631

<a href="https://gitpod.io/#https://github.com/nf-core/nf-co.re/pull/1716"></a>

### CODE REVIEW

File: markdown/developers/contributing_to_pipelines.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+## Before you start
+
+So, you want to add a new pipeline to nf-core - brilliant!
+Before you start typing, check that you're happy with the following points:
+
+- You're familiar with nf-core and nextflow (see our [introduction docs](/docs/usage/introduction.md)).
+- You're used to working with `git` and [GitHub](https://github.com)
+  (see a [nice tutorial here](https://blog.scottlowe.org/2015/01/27/using-fork-branch-git-workflow/))
+- The workflow you're thinking of meets the [nf-core guidelines](https://nf-co.re/docs/contributing/guidelines).
+
```
Comment (christopher-hakkaart) [inline markdown/developers/contributing_to_pipelines.md:15]:
```suggestion
```

File: markdown/developers/contributing_to_pipelines.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+The main steps involved in adding a new nf-core pipeline covered below are:
+
+1. [Joining the community](#join-the-community)
+2. [Contribution overview](#contribution-overview)
+3. [Testing](#testing)
+4. [Patching bugs](#patching-bugs)
+5. [Getting help](#getting-help)
+6. [Pipeline contribution conventions](#pipeline-contribution-conventions)
+
```
Comment (christopher-hakkaart) [inline markdown/developers/contributing_to_pipelines.md:25]:
```suggestion
```

**Review by christopher-hakkaart — CHANGES_REQUESTED** [review]

File: markdown/developers/contributing_to_pipelines.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+## Contribution overview
+
+**⚠️ It's good to introduce your idea early on so that it can be discussed, before you spend lots of time coding. ⚠️**
+
+If you'd like to write some code for an nf-core pipeline, the standard workflow is as follows:
+
+1. Check that there isn't already an issue about your idea for that pipeline to avoid duplicating work. If there isn't one already, please create one so that others know you're working on this
+2. [Fork](https://help.github.com/en/github/getting-started-with-github/fork-a-repo) the the pipeline to your GitHub account
+3. Make the necessary changes / additions on the `dev` branch by using [git checkout](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/committing-changes-to-a-pull-request-branch-created-from-a-fork) within your forked repository following [Pipeline conventions](#pipeline-contribution-conventions)
```
[the remark this answers is not in the document]
Comment (gregorysprenger) [reply markdown/developers/contributing_to_pipelines.md:41]: For additions to a pipeline, you want to clone the repo and use/make changes on the `dev` branch. I think explicitly stating the dev branch will help with issues.

**Comment by maxulysse** [comment]
overwrite https://github.com/nf-core/nf-co.re/pull/1715

**Review by christopher-hakkaart — APPROVED** [review]
- Made some minor technical edits. Looks good otherwise.

**Pull request merged**